### DISCUSSION TITLE
Q: How many lines of C99 code raylib written in?

### DISCUSSION THREAD

**Discussion by ghost** [raysan5/raylib#1633] [Q&A] [answered]
Nice question liked to ask, Feel free to answer it @raysan5 !

**Comment by raysan5** [comment, marked as answer]
Actually, it's a bit difficult to get a reliable measure. I mean, you can measure the lines of code with, for example, [cloc tool](https://github.com/AlDanial/cloc), but those measures do not reflect many aspects of the library. Keep in mind that most of the work done in raylib along the year has not been adding many new features but reviewing and improving the internal codebase. Best improvements usually reduced the line count. 

_github/raysan5/raylib/src_ - Including all _external_ libraries (depending on platform and config, some are not used).
```
github.com/AlDanial/cloc v 1.88  T=2.69 s (47.6 files/s, 83034.8 lines/s)
-----------------------------------------------------------------------------------
Language                         files          blank        comment           code
-----------------------------------------------------------------------------------
C/C++ Header                        64          20380          24858         125147
C                                   42           7040           5236          34799
Objective-C                          5            705            331           2958
make                                 1             65            204            435
Markdown                             2             55              0            409
HTML                                 1             42              4            281
CMake                               11             35             31            146
Windows Resource File                2              2              7             45
-----------------------------------------------------------------------------------
SUM:                               128          28324          30671         164220
-----------------------------------------------------------------------------------
```

_github/raysan5/raylib/src_ - Only base code files

```
github.com/AlDanial/cloc v 1.88  T=0.21 s (118.5 files/s, 178002.8 lines/s)
-----------------------------------------------------------------------------------
Language                         files          blank        comment           code
-----------------------------------------------------------------------------------
C                                    8           3634           3161          14665
C/C++ Header                        13           2241           2682          10079
make                                 1             65            204            435
HTML                                 1             42              4            281
Windows Resource File                2              2              7             45
-----------------------------------------------------------------------------------
SUM:                                25           5984           6058          25505
-----------------------------------------------------------------------------------
```

As you can see, raylib is a very thin layer over base technologies like OpenGL or system libraries. 

Also note that those measures do not include all the examples, templates, build configurations or games created for raylib.

**Reply by ghost** [reply]
@raysan5 Over 26000 lines of code, That's amazing!
Keep up the great work!

**Reply by chriscamacho** [reply]
my counter gives an estimated cost if you wanted to pay someone to write Raylib - $798,189 !! add in the externals and it just becomes redonculous !

**Reply by raysan5** [reply]
@chriscamacho I imagine you used [scc](https://github.com/boyter/scc) to calculate it but I think that tool does not take into account the years of redesigns and reviews with lot of code added and removed. The tool considers lines of code and complexity but raylib has been designed trying to minimize internal code complexity as much as possible, targeting students and novel programmers. Also worth mentioning the work behind the examples, wiki, web, social networks and all the maintenance involved.

I estimate I've invested to date, at least, about 20000 hours in the project (probably more than that).

**Reply by chriscamacho** [reply]
its not even an estimate really is it? but it can be amusing !
